src/self_driving/planning/seq_tuning/sequence_tuning.cpp

```diff
@@ -0,0 +1,216 @@
+#include "self_driving/planning/seq_tuning/sequence_tuning.h"
+
+#include <map>
+#include <vector>
+
+#include "common/managed_pointer.h"
+#include "loggers/selfdriving_logger.h"
+#include "planner/plannodes/abstract_plan_node.h"
+#include "self_driving/planning/action/create_index_action.h"
+#include "self_driving/planning/action/generators/change_knob_action_generator.h"
+#include "self_driving/planning/action/generators/index_action_generator.h"
+#include "self_driving/planning/pilot_util.h"
+#include "self_driving/planning/seq_tuning/graph_solver.h"
+#include "self_driving/planning/seq_tuning/path_solution.h"
+#include "transaction/transaction_manager.h"
+
+namespace noisepage::selfdriving::pilot {
+
+SequenceTuning::SequenceTuning(const PlanningContext &planning_context,
+                               common::ManagedPointer<selfdriving::WorkloadForecast> forecast,
+                               uint64_t end_segment_index)
+    : planning_context_(planning_context), forecast_(forecast), end_segment_index_(end_segment_index) {
+  transaction::TransactionContext *txn = planning_context_.GetTxnManager()->BeginTransaction();
```

Comment:
I see, I copied this from the constructor of `MonteCarloTreeSearch`, but I agree that there's no need to create a new transaction in the constructor of both classes. Also I think we need to remove the `txn` parameter from `PilotUtil::GetQueryPlans` the argument list since the function doesn't use it. Sorry I missed this in the review for the previous PR #1584.

Reply:
You're actually right! I missed that in #1584 as well. Maybe just fix all of them in this PR?

Reply:
Sure!